src/lib/middleware/origin-middleware.ts

```diff
@@ -31,11 +26,15 @@ export const originMiddleware = ({
             const uaLabel = userAgent
                 ? determineIntegrationSource(userAgent)
                 : 'Other';
-            logger.info('API request', {
-                method: req.method,
-                userAgent: req.headers['user-agent'],
-                origin: getFilteredOrigin(req),
-            });
+
+            if (!flagResolver.isEnabled('originMiddlewareRequestLogging')) {
```

Comment:
Don't you mean this instead? Only log if it's enabled?

```suggestion
            if (flagResolver.isEnabled('originMiddlewareRequestLogging')) {
```

Reply:
Wow great catch, what on earth was I doing here? Confused by considering kill switch perhaps